template/flake.nix

```diff
@@ -0,0 +1,55 @@
+{
+  description = "An example of Napalm with flakes";
+
+  # Nixpkgs / NixOS version to use.
+  inputs.nixpkgs.url = "nixpkgs/master";
+
+  # Import napalm
+  inputs.napalm.url = "github:nix-community/napalm";
+
+  outputs = { self, nixpkgs, napalm }:
+    let
+      # Generate a user-friendly version numer.
+      version = builtins.substring 0 8 self.lastModifiedDate;
+
+      # System types to support.
+      supportedSystems = [ "x86_64-linux" ];
+
+      # Helper function to generate an attrset '{ x86_64-linux = f "x86_64-linux"; ... }'.
+      forAllSystems = f:
+        nixpkgs.lib.genAttrs supportedSystems (system: f system);
+
+      # Nixpkgs instantiated for supported system types.
+      nixpkgsFor = forAllSystems (system:
+        import nixpkgs {
+          inherit system;
+          # Use napalm as your overlay
+          overlays = [ self.overlay napalm.overlay ];
+        });
+
+    in {
+
+      # A Nixpkgs overlay.
+      overlay = final: prev:
+        let
+          # In case you do not want to use overlay, you can use:
+          buildNapalmPackage = (napalm.overlay final prev).napalm.buildPackage;
+          # It would work the same way as final.napalm.buildPackage in this case
+        in {
+          # Example packages
+          hello-world = final.napalm.buildPackage ./hello-world { };
+          hello-world-deps = final.napalm.buildPackage ./hello-world-deps { };
```

Comment:
I think smaller template would be sufficient. Do we need both packages?

Reply:
I just took both of the packages that were in the test folder, I can reduce that to just `hello-world-deps` if you prefer so.

Reply:
Yeah, I think it would be better. IMO, the point of template is to get a starting point and I reckon lot of projects will have a single package so they would have to remove the second one.

Also the simpler the template is, the easier it will be to comprehend and modify.